src/azure-cli/azure/cli/command_modules/storage/operations/fs_directory.py

```diff
@@ -0,0 +1,36 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+"""Custom operations for storage file datalake"""
+
+from azure.cli.core.profiles import ResourceType
+
+
+def exists(cmd, client, timeout=None):
+    from azure.core.exceptions import HttpResponseError
+    try:
+        client.get_directory_properties(timeout=timeout)
+        return True
+    except HttpResponseError as ex:
+        from azure.cli.command_modules.storage.track2_util import _dont_fail_on_exist
+        StorageErrorCode = cmd.get_models("_shared.models#StorageErrorCode",
+                                          resource_type=ResourceType.DATA_STORAGE_FILEDATALAKE)
+        _dont_fail_on_exist(ex, StorageErrorCode.blob_not_found)
+        return False
```

Comment:
Put it outside `except HttpResponseError`  in case of other exceptions ?

Reply:
For other exception we need to throw it.